ndbench-dynamodb-plugins/src/main/java/com/netflix/ndbench/plugin/dynamodb/DynamoDBKeyValue.java

```diff
@@ -54,180 +54,218 @@
  * @author ipapapa
  */
 @Singleton
-@NdBenchClientPlugin("DynamoDBKeyValue")
+@NdBenchClientPlugin("DynamoDBKeyValueWithTableCreation")
 public class DynamoDBKeyValue implements NdBenchClient {
-	private final Logger logger = LoggerFactory.getLogger(DynamoDBKeyValue.class);
-	private static AmazonDynamoDB client;
-	private static DynamoDB dynamoDB;
-	private static AWSCredentialsProvider awsCredentialsProvider;
-	private DynamoDBConfigs config;
-	private static Table table;
+    private final Logger logger = LoggerFactory.getLogger(DynamoDBKeyValue.class);
+    private static AmazonDynamoDB client;
+    private static AWSCredentialsProvider awsCredentialsProvider;
+    private static Table table;
 
-	private DataGenerator dataGenerator;
+    private DynamoDBConfigs config;
+    private DataGenerator dataGenerator;
 
+    /**
+     * Credentials will be loaded based on the environment. In AWS, the credentials
+     * are based on the instance. In a local deployment they will have to provided.
+     */
+    @Inject
+    public DynamoDBKeyValue(AWSCredentialsProvider credential, DynamoDBConfigs config) {
+	this.config = config;
+	// if (System.getenv(NdBenchConstants.DISCOVERY_ENV).equals("AWS")) {
```

Comment:
The problem is that `NdBenchConstants.DISCOVERY_ENV` is only used at `ndbench-web` which is not used internally. Hence that variable is not defined in our instances. Hence, this will cause an exception. I have fixed it by checking if it is null or if it is defined and AWS to go through the on instance credentials. However, this will contradict with the OSS `ndbench-web` which assumes that if it is `null` then use locally provided credentials.